### PR TITLE
feat: add update password rotation cmdlet

### LINKED ISSUE #104: Add support for an `Update-PasswordRotationPolicy` cmdlet
### Code of Conduct

- [X] I have read and agree to the project's Code of Conduct.
- [X] Vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue initial description to help the maintainers prioritize.
- [X] Do not leave "+1" or other comments that do not add relevant information or questions.
- [X] If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Description

In v1.4.0, we introduced two new cmdlets:

- `Request-PasswordRotationPolicy`
- `Publish-PasswordRotationPolicy`

This enhancement request will track an additional approvement for the addition of an `Update-PasswordRotationPolicy` cmdlet.

The scope of this cmdlet will be update a password rotation policy for an account: `"autoRotatePolicy": {"frequencyInDays": 90,}`

- disabled
- 30
- 60
- 90
- custom

JSON for a credential's rotation policy. If disabled, the `autoRotatePolicy` is not set.

```json
[
  {
    "id": "e5e1f174-a29a-4d29-b6ba-6a14abde23c3",
    "credentialType": "API",
    "username": "admin",
    "expiry": {
      "expiryDate": "2023-10-04T00:00:00Z",
      "lastCheckedDate": "2023-08-31T01:01:30.35Z",
      "connectivityStatus": "ACTIVE",
      "status": "ACTIVE"
    },
    "resource": {
      "resourceName": "sfo-w01-nsx01.sfo.rainpole.io",
      "resourceType": "NSXT_MANAGER",
      "domainName": "sfo-w01"
    },
    "autoRotatePolicy": {
      "frequencyInDays": 90,
      "nextSchedule": "2023-11-21T00:00:29.133Z"
    }
}
```

Note: This **may** require an update to PowerVCF for a `Set-VcfCredentialExpiry` cmdlet based on the API spec.

### Use Case(s)

As a platform engineer, I need to be able to programmatically update the password rotation for a credential managed by SDDC Manager.

### Potential Configuration

To be determined.

### References

[Add support to remediate password expiration and rotation issues](https://github.com/vmware/powershell-module-for-vmware-cloud-foundation/issues/218)

### PR DESCRIPTION
**Pull request by tenthirtyam**
### Summary

- Added `Update-PasswordRotationPolicy` cmdlet to update the password rotation settings for credentials managed by SDDC Manager.
- Updated `PowerVCF` from v2.3.0 to v2.4.0.
- Updated module version and build to v1.450.1000.
- Updated `CHANGELOG.md`.

### Type

- [ ] Bugfix
- [x] Enhancement or Feature
- [ ] Code Style or Formatting
- [ ] Documentation
- [ ] Refactoring
- [ ] Chore
- [ ] Other
        Please describe:

### Breaking Changes?

- [ ] Yes, there are breaking changes.
- [x] No, there are no breaking changes.

### Test and Documentation

- [x] Tests have been completed.
- [x] Documentation has been added or updated.

Please note that testing requires the use of `PowerVCF` v2.4.0 from the current `develop` branch.

Positive:

```powershell
PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root -autoRotate enabled -frequencyInDays 90
Updating password rotation policy for SSH credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): SUCCESSFUL


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root -autoRotate enabled -frequencyInDays 90
WARNING: Updating password rotation policy for SSH credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): SKIPPED


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root -autoRotate enabled -frequencyInDays 120
Updating password rotation policy for SSH credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): SUCCESSFUL


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root -autoRotate enabled -frequencyInDays 120
WARNING: Updating password rotation policy for SSH credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): SKIPPED


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root -autoRotate disabled
Updating password rotation policy for SSH credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): SUCCESSFUL
```

Negative:

```powershell
PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root2 -autoRotate disabled
Update-PasswordRotationPolicy: Locating password rotation policy for SSH credential (root2) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): PRE_VALIDATION_FAILED


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSHSSH -credentialName root -autoRotate disabled
Update-PasswordRotationPolicy: Cannot validate argument on parameter 'credential'. The argument "SSHSSH" does not belong to the set "ssh,api,audit,sso" specified by the ValidateSet attribute. Supply an argument that is in the set and then try the command again.


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential API -credentialName root -autoRotate disabled
Update-PasswordRotationPolicy: Locating password rotation policy for API credential (root) for workload domain (sfo-m01) vCenter Server resource (sfo-m01-vc01.sfo.rainpole.io) managed by SDDC Manager (sfo-vcf01.sfo.rainpole.io): PRE_VALIDATION_FAILED


PS F:\PowerVCF> Update-PasswordRotationPolicy -server sfo-vcf01.sfo.rainpole.io -user administrator@vsphere.local -pass VMw@re1! -domain sfo-m01 -resource vcenterServer -resourceName sfo-m01-vc01.sfo.rainpole.io -credential SSH -credentialName root2 -autoRotate disabled -frequencyInDays 365
Update-PasswordRotationPolicy: Cannot validate argument on parameter 'frequencyInDays'. The " $autoRotate -eq 'ENABLED' -or $_ -eq $null " validation script for the argument with value "365" did not return a result of True. Determine why the validation script failed, and then try the command again.
```

### Issue References

Closes #104

### CODE REVIEW

File: VMware.CloudFoundation.PasswordManagement.psm1 [hunk 1]
```diff
@@ -8989,6 +8995,136 @@ Function Request-PasswordRotationPolicy {
...
+
+        .PARAMETER credentialName
+        The name of the credential to retrieve the user password rotation settings for.
+
+        .PARAMETER autoRotate
+        Enable or disable the credential password rotation for the credential by SDDC Manager. One of: enabled, disabled.
+
+        .PARAMETER frequencyInDays
+        The number of days of warning before credential's password will be automatically rotated by SDDC Manager.
+        #>
```
Comment (GaryJBlake) [inline VMware.CloudFoundation.PasswordManagement.psm1:9047]: Does this have an extra tab?

**Review by GaryJBlake — APPROVED** [review]
LGTM!

File: VMware.CloudFoundation.PasswordManagement.psm1 (hunk 1 above)
Comment (tenthirtyam) [reply VMware.CloudFoundation.PasswordManagement.psm1]: Amended in https://github.com/vmware/powershell-module-for-vmware-cloud-foundation-password-management/commit/8c963dffbbdd3e5938dc17bb3651b811f8963c40. 😄

**Review by GaryJBlake — APPROVED** [review]
LGTM!

**Pull request merged**